Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -83,7 +84,8 @@ void LogGrpcCallFinished(absl::string_view rpc_name,
                      AsyncQueue *worker_queue,
                      CredentialsProvider *credentials,
                      FSTSerializerBeta *serializer)
-    : grpc_connection_{database_info, worker_queue, &grpc_queue_},
+    : grpc_connection_{database_info, worker_queue, &grpc_queue_,
+                       ConnectivityMonitor::Create(worker_queue)},
```

Comment:
Consider passing this in rather than constructing it here. That way our tests can use a noop or a fake.

Reply:
That's an interesting idea. I'd be inclined to have two versions of the constructor, one taking a `GrpcConnection` (for the sake of tests), and one constructing it in place, like here. I'm not sure about moving construction of `GrpcConnection` for the normal case into `FirestoreClient` -- it would require the client to own `grpc::CompletionQueue`, while my idea was to hide gRPC-related details inside `Datastore`. 

I have a half-baked follow-up with test refactoring, I'll try out this change in that PR.

Reply:
FWIW, I don't think hiding the gRPC gunk from FirestoreClient is all that high priority of a goal. FirestoreClient already owns most of the constellation of objects required to run the client so doing so differently for this case doesn't really help reduce the net complexity of the system.